softwareComponents/stm32cxx/src/drivers/i2c.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <functional>
+
+#include <system/assert.hpp>
+
+#include <drivers/peripheral.hpp>
+#include <drivers/gpio.hpp>
+
+#include <stm32g0xx_ll_i2c.h>
+
+
+// TODO: rework Pin types to be avoid being interchangable
+struct SdaPin : public Gpio::Pin
+{
+    SdaPin( Gpio::Pin pin )
```

Comment:
Ten konstruktor je možné řešit i bez "extra psaní": https://godbolt.org/z/G89xGcveG